cli/src/commands/sign.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap_complete::ArgValueCandidates;
+use itertools::Itertools;
+use jj_lib::commit::CommitIteratorExt;
+use jj_lib::signing::SignBehavior;
+
+use crate::cli_util::CommandHelper;
+use crate::cli_util::RevisionArg;
+use crate::command_error::user_error;
+use crate::command_error::CommandError;
+use crate::complete;
+use crate::ui::Ui;
+
+/// Cryptographically sign a revision
+#[derive(clap::Args, Clone, Debug)]
+pub struct SignArgs {
+    /// What key to use, depends on the configured signing backend.
+    #[arg()]
+    key: Option<String>,
+    /// What revision(s) to sign
+    #[arg(
+        long, short,
+        default_value = "@",
```

Comment:
nit: it might be better to make the default configurable. I think `jj sign` will be used in the context where we do `jj fix && jj sign .. && jj git push`.

(can be addressed separately)

Reply:
Kinda yes. I have no idea whether `revsets.sign` should default to `@` or `reachable(@, mutable())`. I don't think it makes sense to sign the working-copy commit, so the latter might be a better default?

For `unsign`, I wouldn't add a default or configurable revset. I don't expect the user would want to casually remove signatures from a certain set of commits.

In any case, this isn't a blocker. We can add a revset configuration as a follow up patch.

Reply:
> Kinda yes. I have no idea whether revsets.sign should default to @ or reachable(@, mutable()). I don't think it makes sense to sign the working-copy commit, so the latter might be a better default?

`reachable(@, mutable())` does feel better than `@`.

> For unsign, I wouldn't add a default or configurable revset. I don't expect the user would want to casually remove signatures from a certain set of commits.

Yeah, that makes more sense.

> In any case, this isn't a blocker. We can add a revset configuration as a follow up patch.

Let me see how much effort it is to include that. If I am starting to block myself, because I have no idea what I am doing, I will address it in a follow-up.

Reply:
For the sake of getting this merged rather sooner than later, I will address that in #5428.